src/lasdi/workflow.py

```diff
@@ -151,14 +153,18 @@ def initialize_trainer(config, restart_file=None):
 
     physics = initialize_physics(config, param_space.param_name)
     latent_space = initialize_latent_space(physics, config)
+
     if (restart_file is not None):
         latent_space.load(restart_file['latent_space'])
 
     # do we need a separate routine for latent dynamics initialization?
     ld_type = config['latent_dynamics']['type']
     assert(ld_type in config['latent_dynamics'])
     assert(ld_type in ld_dict)
-    latent_dynamics = ld_dict[ld_type](latent_space.n_z, physics.nt, config['latent_dynamics'])
+
+    # Updating the dynamics callback to account for the higher order terms and other non linear functions from the .yml file. 
+
+    latent_dynamics = ld_dict[ld_type](latent_space.n_z, config['latent_dynamics'][ld_type]['higher_order_terms'], config['latent_dynamics'][ld_type]['extra_functions'], physics.nt, config['latent_dynamics'])
```

Comment:
As I mentioned above, `high_order_terms` and `rand_functions` can be parsed from `config` and need not be passed as input arguments. That way we wouldn't have to change this line.